depends/packages/qt.mk

```diff
@@ -125,6 +125,11 @@ $(package)_config_opts_darwin += -device-option MAC_SDK_VERSION=$(OSX_SDK_VERSIO
 $(package)_config_opts_darwin += -device-option CROSS_COMPILE="$(host)-"
 $(package)_config_opts_darwin += -device-option MAC_TARGET=$(host)
 $(package)_config_opts_darwin += -device-option XCODE_VERSION=$(XCODE_VERSION)
+else
+ifneq ($(host_arch),$(build_arch))
```

Comment:
Haven't looked at this other than to say, if this is explicitly fixing the `NO_QT=1` build, why is it touching Qt code.

Reply:
The reason to include this change here is to do not change build system behavior while building without `NO_QT=1`:
- on Intel-based macOS 11.4
- on M1-based macOS with installed Rosetta 2

Reply:
the "darwin" comes out of the blue here, it's not in a MacOS specific block, `($(host_arch),$(build_arch))` is not specifically related to it and might arise on different systems as well when cross-compiling

Reply:
These lines are placed within the `else` branch of the `ifneq ($(build_os),darwin)`.